### ISSUE TITLE
How to solve issue with ajaxProgressiveLoad: Scroll and ajaxSorting?

### ISSUE THREAD

**Issue by Alarich** [olifolkerd/tabulator#1400]
Greetings!

Firstly I'd like to thank @olifolkerd for creating this wonderful tool and also the contributors. 

Now, onto the issue I'm having.

I've come upon a situation where I need to use ajaxProgressiveLoad: scroll and ajaxSorting together, but they seem to be incompatible. 
The same issue has been queried in #1092, but never found a solution.

My expectation would be, that ajaxSorting would set the page number to 1 and start the ajaxProgressiveLoad process again with the new sort parameters.

The main issue, without going too deep into the library, seems to be that ajaxProgressiveLoad: scroll requires the data to be returned in this form:
```
current_page: 1,
data: [{item1},{item2}],
last_page:50,
```

ajaxSorting, however, requires the result to be in this form:
```
[{item1},{item2}]
```

Now, I've managed to kind of get around this issue by utilizing the ajaxResponse callback:
```
ajaxResponse: function(url, params, response) {
    if (params.sorters && params.sorters.length > 0 ) {
        return response.data;
    }
    return response;
}
```

But this has two issues:
1. Since sorters are set, ajaxProgressiveLoad: scroll will no longer work, as it expects the first response and even though the params object appears to have the page property unset, accessing it will return undefined, as it seems to be filled after the ajaxResponse callback. This leaves no method for checking if the page variable actually exists or not (it wouldn't on the ajaxSorting request).
2. There doesn't appear to be a clear way to reset the page number to 1 for ajaxProgressiveLoad: scroll.

I could probably bypass this issue on the server side, simply check that the page variable has not been sent and just return the data contents. 
However, that still leaves me stumped on how to set the page to 1, as calling `$('#datatable_table_contact_list').tabulator("setPage",1);` simply returns false and does not reset the page number.

I'll probably figure this out eventually, but any guidance is appreciated.

**How have others solved the issue with AjaxProgressiveLoad:Scroll and ajaxSorting?**

Just to be prudent, I will include my configuration as well:

```


    document.addEventListener('DOMContentLoaded', function() {

        var selector = "#datatable_table_contact_list";
        var oTable;

        var defaults = {
        };

        var columns = {
            columns:[
              {
                title:"No.",
                field:"contact_list_number",
                headerSort:true,
              },
              {
                title:"Name\u0020\u007C\u0020Address",
                field:"contact_list_info",
                headerSort:true,
                formatter:"html",
              },
              {
                title:"Country",
                field:"contact_list_country",
                headerSort:true,
              },
              {
                title:"Registry\/ID",
                field:"contact_list_registry",
                headerSort:false,
              },
              {
                title:"Phone",
                field:"contact_list_phone",
                headerSort:false,
              },
              {
                title:"Email",
                field:"contact_list_email",
                headerSort:false,
              },
              {
                title:"Last\u0020activity",
                field:"contact_list_last_activity",
                headerSort:true,
              },
              {
                title:"Manager",
                field:"contact_list_manager",
                headerSort:true,
              },
              {
                title:"Status",
                field:"contact_list_status",
                headerSort:false,
              },
            ]
        };

        var options = {
          height:'311px',
          placeholder:'No\u0020data\u0020set',
          layout:'fitColumns',
          ajaxURL:'\/backend\/contact\/',
          ajaxConfig:'POST',
          ajaxFiltering:'1',
          ajaxSorting:'1',
          ajaxProgressiveLoad:'scroll',
          ajaxProgressiveLoadDelay:'10',
          ajaxProgressiveLoadScrollMargin:'400',
          paginationSize:'20',
        };
        
        var callbacks = {
          "paginationDataSent":{
            "page":"page",
            "size":"perPage",
            "sort":"sort",
            "sort_dir":"sortDir",
            "filter":"filter",
            "filter_value":"filterValue",
            "filter_type":"filterType",
            "sorters":"sorters",
          },
          "ajaxResponse":function(url, params, response){
            //url - the URL of the request
            //params - the parameters passed with the request
            //response - the JSON object returned in the body of the response.

            if (params.sorters && params.sorters.length > 0) {
                return response.data; //return the data property of a response json object
            }
            return response;
          },
        };

        function buildDatatablecontact_list() {
            $.extend(defaults, columns);
            $.extend(defaults, options);
            $.extend(defaults, callbacks);
            oTable = $(selector)
                    .tabulator(defaults);
        }

        buildDatatablecontact_list();

    });
```

**Comment by olifolkerd** [comment]
Hey @Alarich 

Thanks for getting in touch.

This is a known bug and has been raised as an issue several times. there is a incompatibility in v3.5 between the progressive loading and ajax filtering/sorting.

You will be happy to hear that this issue has been resolved in version 4.0, which is currently undergoing a public beta, and will be included in the official release next weekend.

Cheers

Oli :)

**Comment by Alarich** [comment]
Thank you @olifolkerd.

I will move over to the new version then :)
